src/Workspaces/CoreTestUtilities/ExportProviderCache.cs

```diff
@@ -52,6 +72,22 @@ public static ComposableCatalog CreateAssemblyCatalog(Assembly assembly)
 
         public static ComposableCatalog CreateAssemblyCatalog(IEnumerable<Assembly> assemblies, Resolver resolver = null)
         {
+            if (assemblies is ImmutableArray<Assembly> assembliesArray)
+            {
+                if (s_defaultHostCatalog != null && assembliesArray == MefHostServices.DefaultAssemblies)
+                {
+                    return s_defaultHostCatalog;
+                }
+                else if (s_desktopHostCatalog != null && assembliesArray == DesktopMefHostServices.DefaultAssemblies)
```

Comment:
Oof this is magic: so the static initializer calls the method that then uses the field inside of it? That seems like it's ripe for breaking.

Reply:
➡️ Refactoring to avoid magic